src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -386,14 +446,86 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
     /// ```
     /// For the above schema, both `fields` and `type` constraints map to different abstract data types
     /// respectively Struct(with given fields `source` and `destination`) and Value(with a single field that has String data type).
-    fn verify_abstract_data_type_consistency(
+    fn verify_and_update_abstract_data_type(
         &mut self,
         current_abstract_data_type: AbstractDataType,
+        tera_fields: &mut Vec<Field>,
         code_gen_context: &mut CodeGenContext,
     ) -> CodeGenResult<()> {
         if let Some(abstract_data_type) = &code_gen_context.abstract_data_type {
-            if abstract_data_type != &current_abstract_data_type {
-                return invalid_abstract_data_type_error(format!("Can not determine abstract data type as current constraint {} conflicts with prior constraints for {}.", current_abstract_data_type, abstract_data_type));
+            match abstract_data_type {
+                // In the case when a `type` constraint occurs before `element` constraint. The element type for the sequence
+                // needs to be updated based on `element` constraint whereas sequence type will be used as per `type` constraint.
+                // e.g. For a schema as below:
```

Comment:
It seems like there's a fair amount of complexity here that's dedicated to accommodating the ISL fields appearing in unpredictable orders. I think we could easily sort the fields beforehand so we could process them more easily, guaranteeing that we see `name` before `type` before `element`, etc.

Reply:
Thats a good suggestion. I will create an issue for it and work on it separately.

Reply:
Reference issue: https://github.com/amazon-ion/ion-cli/issues/108